orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/utils/TrafficGuard.java

```diff
@@ -43,19 +43,56 @@ public TrafficGuard(OortHelper oortHelper, Front50Service front50Service) {
   }
 
+  public void verifyInstanceTermination(List<String> instanceIds, String account, Location location, String cloudProvider, String operationDescriptor) {
+    Map<String, List<String>> instancesPerServerGroup = new HashMap<>();
+    instanceIds.forEach(instanceId -> {
+
+      String resolvedServerGroupName = resolveServerGroupNameForInstance(instanceId, account, location.getValue(), cloudProvider);
+      if (resolvedServerGroupName != null) {
+        instancesPerServerGroup.computeIfAbsent(resolvedServerGroupName, serverGroup -> new ArrayList<>()).add(instanceId);
+      }
+    });
+
+    instancesPerServerGroup.entrySet().forEach(entry -> {
+      String serverGroupName = entry.getKey();
+      Names names = Names.parseName(serverGroupName);
+      if (hasDisableLock(names.getCluster(), account, location)) {
+        Optional<TargetServerGroup> targetServerGroup = oortHelper.getTargetServerGroup(account, serverGroupName, location.getValue(), cloudProvider);
+
+        targetServerGroup.ifPresent(serverGroup -> {
+          long otherActiveInstances = serverGroup.getInstances().stream().filter(i -> "Up".equals(i.get("healthState")) && !entry.getValue().contains(i.get("name"))).count();
+          if (otherActiveInstances == 0) {
+            verifyOtherServerGroupsAreTakingTraffic(serverGroupName, location, account, cloudProvider, operationDescriptor);
+          }
+        });
+      }
+    });
+  }
+
+  private String resolveServerGroupNameForInstance(String instanceId, String account, String region, String cloudProvider) {
+    List<Map> searchResults = (List<Map>) oortHelper.getSearchResults(instanceId, "instances", cloudProvider).get(0).getOrDefault("results", new ArrayList<>());
+    Optional<Map> instance = searchResults.stream().filter(r -> account.equals(r.get("account")) && region.equals(r.get("region"))).findFirst();
+    // instance not found, assume it's already terminated, what could go wrong
+    return (String) instance.orElse(new HashMap<>()).get("serverGroup");
+  }
+
   public void verifyTrafficRemoval(String serverGroupName, String account, Location location, String cloudProvider, String operationDescriptor) {
     Names names = Names.parseName(serverGroupName);
 
     if (!hasDisableLock(names.getCluster(), account, location)) {
       return;
     }
 
+    verifyOtherServerGroupsAreTakingTraffic(serverGroupName, location, account, cloudProvider, operationDescriptor);
+  }
+
+  private void verifyOtherServerGroupsAreTakingTraffic(String serverGroupName, Location location, String account, String cloudProvider, String operationDescriptor) {
+    Names names = Names.parseName(serverGroupName);
     Optional<Map> cluster = oortHelper.getCluster(names.getApp(), account, names.getCluster(), cloudProvider);
 
     if (!cluster.isPresent()) {
-      throw new IllegalStateException("Could not find traffic-protected cluster.");
+      throw new IllegalStateException("Could not find cluster with traffic guard configured.");
```

Comment:
IMO let's throw some context into the exception message (server group name, etc).

Similar on the exception below.